wgsl/index.bs

```diff
@@ -6717,14 +6717,25 @@ When an identifier is used as a [=syntax/callable=] item, it is one of:
 
     | [=syntax/ident=]
 
-    | [=syntax/callable=] [=syntax/argument_expression_list=]
+    | [=syntax/call_expression=]
 
     | [=syntax/literal=]
 
     | [=syntax/paren_expression=]
 
     | <a for=syntax_kw lt=bitcast>`'bitcast'`</a> <a for=syntax_sym lt=less_than>`'<'`</a> [=syntax/type_specifier=] <a for=syntax_sym lt=greater_than>`'>'`</a> [=syntax/paren_expression=]
 </div>
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>call_expression</dfn> :
+
+    | [=syntax/call_phrase=]
```

Comment:
I'm not sure I follow the need for this indirection. What's the downside of eliminating call_phrase and making the statement directly a call_expression?

Reply:
(I also think in a similar vein and keep the structure duplicated between stmt and expr)

Reply:
Then by the rules it requires the type checker to derive a type for a call to a control barrier:

```
fn foo() {
   // if the grammar matching goes through call_expression then the type checker
   // will attempt to derive a type for the return value of this function.
   workgroupBarrier();   
}
```

Reply:
I see and we don't have void.